File: app/src/main/java/com/ponthaitay/lifecycleactivity/googleio2017/room/javacode/MainRoomActivity.java
package com.ponthaitay.lifecycleactivity.googleio2017.room.javacode;

import android.annotation.SuppressLint;
import android.arch.lifecycle.Observer;
import android.arch.lifecycle.ViewModelProviders;
import android.arch.persistence.room.Room;
import android.os.Bundle;
import android.util.Log;

import com.ponthaitay.lifecycleactivity.googleio2017.R;

import java.util.List;

import io.reactivex.annotations.Nullable;

public class MainRoomActivity extends BaseLifecycleActivity {

    @SuppressLint("StaticFieldLeak")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main_room);

        final StudentDatabase db = Room.databaseBuilder(this, StudentDatabase.class,
                "studentdatabase").build();

        final StudentEntity entity = new StudentEntity();
        entity.setId(1);
        entity.setFirstName("Test");
        entity.setLastName("test");

        StudentViewModel studentViewModel = ViewModelProviders.of(this).get(StudentViewModel.class);
        studentViewModel.insertStudent(db.studentDao(), entity);
        studentViewModel.getStudentById(db.studentDao(), 1).observe(this, new Observer<List<StudentEntity>>() {
            @Override
            public void onChanged(@Nullable List<StudentEntity> studentEntities) {
                if (!studentEntities.isEmpty()) Log.d("POND", String.valueOf(studentEntities.get(0).getFirstName()));
                else Log.d("POND", "error");
            }
        });
    }
}

File: app/src/main/java/com/ponthaitay/lifecycleactivity/googleio2017/room/kotlincode/MainRoomActivity.kt
package com.ponthaitay.lifecycleactivity.googleio2017.room.kotlincode

import android.arch.lifecycle.Observer
import android.arch.lifecycle.ViewModelProviders
import android.arch.persistence.room.Room
import android.os.Bundle
import android.util.Log
import com.ponthaitay.lifecycleactivity.googleio2017.R


class MainRoomActivity : BaseLifecycleActivity() {

    val TAG: String = javaClass.simpleName
    var employeeDB: EmployeeDatabase? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main_room)
        employeeDB = Room.databaseBuilder(this, EmployeeDatabase::class.java, "employee_db").build()
        val dao = employeeDB?.employeeDao()

        val jedsadaEmpolyee = EmployeeEntity()
        jedsadaEmpolyee.firstName = "test"
        jedsadaEmpolyee.lastName = "test"

        val employeeModel = ViewModelProviders.of(this).get(EmployeeViewModel::class.java)

        employeeModel.insertStudent(dao, jedsadaEmpolyee)
        employeeModel.getStudentById(dao, jedsadaEmpolyee.firstName)
                .observe(this, Observer { setResultQueryEmployee(it) })
    }

    fun setResultQueryEmployee(list: List<EmployeeEntity>?) {
        Log.d(TAG, "$list")
    }
}